tests/python/gaia-ui-tests/gaiatest/tests/functional/homescreen/test_homescreen_status_bar_icons_visibility.py

```diff
@@ -6,27 +6,38 @@
 
 from gaiatest import GaiaTestCase
 from gaiatest.apps.system.app import System
+from gaiatest.apps.homescreen.app import Homescreen
 
 
 class TestHomescreenStatusBarIconsVisibility(GaiaTestCase):
 
-    homescreen_frame_locator = (By.CSS_SELECTOR, '#homescreen iframe')
-    homescreen_all_icons_locator = (By.CSS_SELECTOR, 'gaia-grid .icon')
+    scollable_div_locator = (By.CSS_SELECTOR, '#apps-panel div.scrollable')
 
     def setUp(self):
         GaiaTestCase.setUp(self)
+        # Set the time on the device so that it starts with 0 seconds
+        _seconds_since_epoch = self.marionette.execute_script("""
+                var today = new Date();
+                var yr = today.getFullYear();
+                var mth = today.getMonth();
+                var day = today.getDate();
+                return new Date(yr, mth, day, 1, 0, 0).getTime();""")
+        self.data_layer.set_time(_seconds_since_epoch)
 
     def test_homescreen_status_bar_icons_visibility(self):
+        screenshot1 = self.marionette.screenshot(System(self.marionette).status_bar.root_element)
+        self.apps.switch_to_displayed_app()
+        homescreen = Homescreen(self.marionette)
+        app_icons = homescreen.visible_apps
+        last_icon = app_icons[len(app_icons)-1]
         # scroll last icon into homescreen view
-        self.marionette.switch_to_frame(
-            self.marionette.find_element(*self.homescreen_frame_locator))
-        homescreen_last_icon = self.marionette.find_elements(
-            *self.homescreen_all_icons_locator)[-1]
         self.marionette.execute_script(
-            'arguments[0].scrollIntoView(false);', [homescreen_last_icon])
-        self.assertGreater(self.marionette.execute_script(
-            "return window.scrollY"), 0)
+            'arguments[0].scrollIntoView(true);', [last_icon.root_element])
```

Comment:
The old test was like this already, but I think we can improve the readiness, here:
```py
last_icon_position = len(app_icons)-1
homescreen.scroll_to_icon(icon_position=last_icon_position)
```